### ISSUE TITLE
`dotnet` 8 tool crashes in WSL under Windows ARM64 VM

### ISSUE THREAD

**Issue by roycornelissen** [dotnet/sdk#37261]
### Describe the bug

We may have a relatively exotic situation but some of our devs are running a Windows ARM64 in Parallels on a Mac M1/M2 processor. In this Windows VM, we run some build scripts and Terraform scripts in Ubuntu 22.04 under WSL1.

We have installed .NET using the install script as described here: https://learn.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#set-environment-variables-system-wide

Up and until .NET 7, this has worked fine. Since we upgraded our application to .NET 8 and after installing the .NET 8.0 SDK, the `dotnet` tool now crashes:

```bash
$ dotnet --version

Fatal error. AdvSimd processor support required.
Aborted (core dumped)
```

### To Reproduce

- Create Windows for ARM VM in Parallels on a Mac M1/M2 machine
- Install Ubuntu 22.04 in WSL1
- Install .NET 8 using the `dotnet-install.sh` script as [described here](https://learn.microsoft.com/en-us/dotnet/core/install/linux-scripted-manual#set-environment-variables-system-wide)
- Run `dotnet` with any command, e.g. `dotnet --version` or `dotnet publish`

### Exceptions (if any)

`dotnet` crashes with the following message:

```bash
Fatal error. AdvSimd processor support required.
Aborted (core dumped)
```

### Further technical details
- .NET 8 installed using `./dotnet-install.sh --channel 8.0 --architecture arm64`
- Installed SDK version: `8.0.100.0`

Cannot output the details of `dotnet --version` because this command crashes.

**Comment by creeperlv** [comment]
Hi, I think I can give some information here.
I am experiencing exact same problem with `dotnet 8` in WSL1, but I am using real Windows 11 on Arm Hardware.
I have two WoA devices, one is Snapdragon 8cx Gen 2 device, the other is Snapdragon 855 device.
However, there is no problem with WSL2, dotnete 8 tool runs completely fine with Debian in WSL2.

**Comment by roycornelissen** [comment]
Yeah I guess my scenario is a bit exotic due to the nested virtualization scenario

I cannot use WSL2 in my Parallels image sadly, but I am able to work around it by doing the build and publish on Windows for now.

Interesting though that you are also experiencing the same in WSL1. 🤔

**Comment by creeperlv** [comment]
> Interesting though that you are also experiencing the same in WSL1. 🤔
I guess the reason is that the kernel of WSL1 does not process SIMD for Arm64 for WSL1 predates Windows 10 on Arm. I experience this for my Snapdragon 855 device does not support virtualization hence, I can only use WSL1 on that device.

**Comment by creeperlv** [comment]
I just verified that it might be related to WSL1 not exposing `asimddp`. I executed `lscpu` command in WSL1 and WSL2. Following are results:
![image](https://github.com/dotnet/sdk/assets/7870727/bfd0ad97-e0dd-45b4-bacf-dbf0f1e1c8a8)

![image](https://github.com/dotnet/sdk/assets/7870727/7015bd25-59ad-479b-8a5d-b3e66cd2ae25)

As output displays, WSL1 lacks a lot of CPU features.

**Comment by nagilson** [comment]
@vitek-karas @agocke  This is not an IL Link issue but we think this might be something for the CLR AppModel Team. May you help route this?

**Comment by agocke** [comment]
.NET does not support wsl1
